sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/hss/HssEntry.kt

```diff
@@ -72,17 +50,25 @@ class SimpleHssAdapter(val httpClient: CloseableHttpClient,
      * @param simEntry  SIM profile to activate
      * @return Updated SIM Entry
      */
-    override fun activate(simEntry: SimEntry): Either<SimManagerError, SimEntry> {
 
-        if (simEntry.iccid.isEmpty()) {
-            return NotUpdatedError("Illegal parameter in SIM activation request to BSSID ${config.name}")
+    override fun activate(hssName: String, iccid: String, msisdn: String): Either<SimManagerError, Unit> {
+        if (hssName != name) {
+            return Left(AdapterError("Attempt to activate  hssName=$hssName, iccid=$iccid, msisdn=$msisdn in a dispatcher for hss named $name"))
+        }
+
+        if (!iccid.matches(Regex("^\\d{19,20}"))) {
```

Comment:
Make convenience function and  use it in more places?

Reply:
I'll let this slide for now.  Added a more elaborate comment so it may. be considered again later.